code/game/sound.dm

```diff
@@ -26,12 +26,14 @@
 		if(get_dist(M, turf_source) <= maxdistance)
 			sound_or_datum(M, turf_source, source, input, vol, vary, frequency, falloff, channel, pressure_affected)
 
-/proc/sound_or_datum(mob/receiver, turf/turf_source, datum/source, input, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
+/proc/sound_or_datum(mob/receiver, turf/turf_source, atom/source, input, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
 	if(istype(input, /datum/outputs))
 		var/last_played_time = source.datum_outputs[input]
 		var/datum/outputs/O = input
 		O.send_info(receiver, turf_source, vol, vary, frequency, falloff, channel, pressure_affected, last_played_time)
-		source.datum_outputs[input] = world.time
+		if(O.cooldown[2] == TRUE) //if we're cooled down ,reset the time
+			source.datum_outputs[input] = world.time
+			O.cooldown[2] = FALSE //start cooling down again
```

Comment:
You check the global type datum output var for existance of a state of cooldown but then write the individual time into the object
all other objects will not only not be able to output text because the global type var was set to be on cooldown but they wont even be able to access the time of when it will expire because despite the cooldown being global the time it will expire is stored in the object that set this cooldown up

Reply:
no it literally doesn't, here im even gonna test it
![image](https://user-images.githubusercontent.com/16103879/53693166-5404a700-3da5-11e9-9010-0294fabb3408.png)
each plush, even those of the same type, have their own individual cooldown, i step on one plush, it gives me a text message and refuses to send text for 10 seconds, but if i step on a plush of the same type during those 10 seconds it outputs text normally and has its own cooldown